contribute cookbook: simple subduction initiation

cedrict opened the pull request:
Pull Request Checklist. Please read and check each box with an X. Delete any part not applicable. Ask on the [forum](https://community.geodynamics.org/c/aspect) if you need help with any step.

*Describe what you did in this PR and why you did it.*

### Before your first pull request:

* [X ] I have read the guidelines in our [CONTRIBUTING.md](../blob/main/CONTRIBUTING.md) document.

### For all pull requests:

* [ X] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).

### For new features/models or changes of existing features:

* [ ] I have tested my new feature locally to ensure it is correct.
* [ ] I have [created a testcase](https://aspect-documentation.readthedocs.io/en/latest/user/extending/testing/writing-tests.html) for the new feature/benchmark in the [tests/](../blob/main/tests/) directory.
* [ ] I have added a changelog entry in the [doc/modules/changes](../blob/main/doc/modules/changes) directory that will inform other users of my change.

jdannberg reviewed (commented):
Thank you for adding this!

I have two general comments in addition to the ones I left in the code: 

(1) Can you add a test case? It can just include your input file (add the line `include $ASPECT_SOURCE_DIR/cookbooks/subduction_initiation/subduction_initiation_compositional_fields.prm`) and then maybe reduce the resolution/end time. This way, we make sure it will continue to work and the results will still be correct in the future. I would actually do this first (before any of the other changes I suggested) because then you can use it as well to make sure that changing the input file does not change your solution. 

(2) This is more a philosophical question: Your water viscosity is 1e19, which is the same as the lowest value of the asthenosphere viscosity (case 6). Do you think that is still a reasonable approximation? Could you maybe add a line to the text on that?

Thank you!

bangerth commented on cookbooks/subduction_initiation/doc/subduction_initiation.md:
```diff
@@ -0,0 +1,91 @@
+
+The setup for this experiment originates in {cite:t}`matsumoto:tomoda:1983`.
+
+In this very early computational geodynamics paper the authors
```
```suggestion
In this very early computational geodynamics paper, the authors
```

cedrict replied:
done

bangerth commented on cookbooks/subduction_initiation/doc/subduction_initiation.md:
```diff
@@ -0,0 +1,91 @@
+at the gigantic fracture zone in the Northeastern
+Pacific by means of numerical simulation
+of contact between two kinds of viscous fluid of different density.
+They then solve the incompressible isothermal linear Stokes equations
```
```suggestion
They then formulate the incompressible isothermal linear Stokes equations
```

bangerth commented on cookbooks/subduction_initiation/doc/subduction_initiation.md:
```diff
@@ -0,0 +1,91 @@
+lithosphere ($\rho_l=3300~\text{ kg m}^{-3}$, $\eta_l$) and
+asthenosphere ($\rho_a=3200~\text{ kg m}^{-3}$, $\eta_a$).
+Note that although the water viscosity is correct, this is probably not
+the value that was used in the authors since it would most likely lead to
```
```suggestion
the value that was used in the paper since it would most likely lead to
```

cedrict replied:
already corrected (see Julianne's comment above)

bangerth commented on cookbooks/subduction_initiation/doc/subduction_initiation.md:
```diff
@@ -0,0 +1,91 @@
+asthenosphere ($\rho_a=3200~\text{ kg m}^{-3}$, $\eta_a$).
+Note that although the water viscosity is correct, this is probably not
+the value that was used in the authors since it would most likely lead to
+numerical errors. We then set $\eta_w=10^{19}~\text{ Pa s}$ (we can then speak of 'sticky water').
```
```suggestion
numerical errors. We thus set $\eta_w=10^{19}~\text{ Pa s}$ (we can then speak of 'sticky water').
```

bangerth replied:
:-)

Honey?

cedrict replied:
sticky honey is a hard sell ... is there a planet covered in honey with plate tectonics?

bangerth reviewed (commented):
I liked the "sticky water" analogy. Perhaps honey?

I had to look up the viscosity of honey, and found it to be on the order of 50 Pa.s. So your water with a viscosity of 1e19 Pa.s is *really* sticky!

One also finds this publication (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4745488/#:~:text=Viscosity%20of%20all%20the%20honey,54.0%20to%2089.0%20kJ%2Fmol.) wherein the authors also see whether irradiation with gamma rays changes the viscosity -- though I will admit that I cannot see why anyone would find themselves interested in the effects of radition on viscosity.

bangerth commented on cookbooks/subduction_initiation/doc/subduction_initiation.md:
```diff
@@ -0,0 +1,91 @@
+the value that was used in the authors since it would most likely lead to
+numerical errors. We then set $\eta_w=10^{19}~\text{ Pa s}$ (we can then speak of 'sticky water').
+Also, all viscosities in the paper
+are expressed in Poise with $1~\text{ Poise}=0.1~\text{ Pa s}$.
```
Ah, come on. Who thought that was a good idea?

bangerth commented on cookbooks/subduction_initiation/subduction_initiation_compositional_fields.prm:
```diff
@@ -0,0 +1,98 @@
+#parameter file for replicating Matsumoto & Tomoda 1983
```
```suggestion
# Parameter file for replicating Matsumoto & Tomoda 1983
```

cedrict replied:
done

cedrict commented on cookbooks/subduction_initiation/doc/subduction_initiation.md:
```diff
@@ -0,0 +1,91 @@
+lithosphere ($\rho_l=3300~\text{ kg m}^{-3}$, $\eta_l$) and
+asthenosphere ($\rho_a=3200~\text{ kg m}^{-3}$, $\eta_a$).
+Note that although the water viscosity is correct, this is probably not
+the value that was used in the authors since it would most likely lead to
```
done

cedrict commented on cookbooks/subduction_initiation/doc/subduction_initiation.md:
```diff
@@ -0,0 +1,91 @@
+asthenosphere ($\rho_a=3200~\text{ kg m}^{-3}$, $\eta_a$).
+Note that although the water viscosity is correct, this is probably not
+the value that was used in the authors since it would most likely lead to
+numerical errors. We then set $\eta_w=10^{19}~\text{ Pa s}$ (we can then speak of 'sticky water').
```
done

cedrict commented on cookbooks/subduction_initiation/doc/subduction_initiation.md:
````diff
@@ -0,0 +1,91 @@
+Taken from {cite:t}`matsumoto:tomoda:1983`.
+```
+
+Two parameter files are provided for Case 1: one using
+compositional fields, and one using the particle-in-cell approach.
````
nice!

cedrict commented on cookbooks/subduction_initiation/doc/subduction_initiation.md:
```diff
@@ -0,0 +1,91 @@
+<img src="results3.*" width="95%" />
+
+Case 1 (obtained with the compositional fields approach) at
+(a) 0 Ma; (b) 12 Ma; (c) 21 Ma; (d) 31 Ma; (e) 41 Ma.
```
added units in caption. also re-uploading figure bc original was too low resolution.

cedrict commented on cookbooks/subduction_initiation/subduction_initiation_particle_in_cell.prm:
```diff
@@ -0,0 +1,104 @@
+set Output directory                       = output-subduction-initiation-pic
+set Pressure normalization                 = surface
+
+#subsection Solver parameters
+#  subsection Stokes solver parameters
+#    set Number of cheap Stokes solver steps = 0
+#  end
+#end
```
done

cedrict commented on cookbooks/subduction_initiation/subduction_initiation_compositional_fields.prm:
```diff
@@ -0,0 +1,98 @@
+  set Number of fields = 4
+end
+
+subsection Initial composition model
+  set Model name = function
+  subsection Function
+    set Variable names      = x,y
+    set Function constants  = L0=300e3
+    set Function expression = if((x>=L0 && y<162e3),1,0) ;\
+                              if((x<=L0 && y>120e3 && y<=170e3),1,0) ;\
+                              if((x>=L0 && y>=162e3 && y<=172e3),1,0) ;\
+                              if((x<=L0 && y>170e3) || (x>=L0 && y>172e3),1,0)
+  end
+end
```
good idea, thx!

cedrict commented on cookbooks/subduction_initiation/doc/subduction_initiation.md:
```diff
@@ -0,0 +1,91 @@
+Two parameter files are provided for Case 1: one using
+compositional fields, and one using the particle-in-cell approach.
+Other cases can be run by changing the viscosities and/or the domain size accordingly in the
+parameter files.
```
done

cedrict commented on cookbooks/subduction_initiation/subduction_initiation_particle_in_cell.prm:
```diff
@@ -0,0 +1,104 @@
+#parameter file for replicating Matsumoto & Tomoda 1983
```
ok

cedrict replied:
have created a 'subduction_initiation_basis.prm' file which contains the common prm code for both.

cedrict commented on cookbooks/subduction_initiation/subduction_initiation_particle_in_cell.prm:
```diff
@@ -0,0 +1,104 @@
+    set Data output format       = vtu
+    set List of particle properties = initial composition, velocity
+    set Particle generator name = random uniform
+    set Interpolation scheme = cell average #default
```
done

cedrict commented:
I have changed the water viscosity to 1e18. It does not change the results shown in the cookbook very much at all. But 1e18 is indeed safer than 1e19 in case 6. I have also added a link to Crameri paper.

cedrict commented:
ready to review again. Still need to do a test

bangerth commented:
This looks good. Do you want us to merge this now, or wait for the test?

cedrict commented:
please merge, am now looking into making a test (could take a while, it's been a looong time :)

bangerth commented:
There's a failing check:
```
--- a/cookbooks/subduction_initiation/subduction_initiation_basis.prm
+++ b/cookbooks/subduction_initiation/subduction_initiation_basis.prm
@@ -46,4 +46,3 @@ subsection Mesh refinement
   set Strategy                      = composition
   set Coarsening fraction           = 0
 end
-
```
Perhaps an empty line at the end of a file?

Other than that, good to merge!
